subsys/debug/thread_analyzer/thread_analyzer.c

```diff
@@ -11,6 +11,7 @@
 #include <zephyr/kernel.h>
 /* For z_stack_space_get() */
 #include <kernel_internal.h>
+#include <ksched.h>
```

Comment:
any inclusion of this file outside of the kernel/ is technical_debt++, this probably need to be implemented as part of k_thread_runtime_stats API instead and not using private function declarations. 

@peter-mitsis  please take a look at this and suggest an alternative solution.

Reply:
As I understand it, the inclusion of ksched.h is so that we can get access to the newly proposed z_sched_thread_reset_long_frame(). I'm OK for giving this a public name such as k_sched_thread_reset_longest_frame(). Someone is likely going to want that capability besides the thread_analyzer.

Reply:
so is this moving something you'd want us to do as part of the PR?  Or are we holding until that move can be completed?

Reply:
alright I believe this is now done.  Please take a look

Reply:
ok, so now we are including internal kernel headers and use the not-so-welcome newly introduced z_ prefixed functions, still not great. @peter-mitsis  can't this be converted to a public API? to make things a bit future proof?